ethereumj-core/src/main/java/org/ethereum/net/eth/handler/Eth62.java

```diff
@@ -517,58 +520,52 @@ protected synchronized void sendNextHeaderRequest() {
 
     protected synchronized void processInitHeaders(List<BlockHeader> received) {
 
-        BlockHeader first = received.get(0);
+        final BlockHeader blockHeader = received.get(0);
+        final long blockNumber = blockHeader.getNumber();
+
         if (ethState == EthState.STATUS_SENT) {
-            updateBestBlock(first);
+            updateBestBlock(blockHeader);
 
             logger.trace("Peer {}: init request succeeded, best known block {}",
                     channel.getPeerIdShort(), bestKnownBlock);
 
             // checking if the peer has expected block hashes
             ethState = EthState.HASH_CONSTRAINTS_CHECK;
 
-            List<Pair<Long, byte[]>> constraints = config.getBlockchainConfig().
-                    getConfigForBlock(first.getNumber()).blockHashConstraints();
-
-            blockHashCheck = Collections.synchronizedMap(new HashMap<Long, byte[]>());
-            for (Pair<Long, byte[]> constraint : constraints) {
-                if (constraint.getLeft() <= getBestKnownBlock().getNumber()) {
-                    blockHashCheck.put(constraint.getLeft(), constraint.getRight());
+            validatorMap = Collections.synchronizedMap(new HashedMap());
+            List<Pair<Long, BlockHeaderValidator>> validators = config.getBlockchainConfig().
+                    getConfigForBlock(blockNumber).headerValidators();
+            for (Pair<Long, BlockHeaderValidator> validator : validators) {
+                if (validator.getLeft() <= getBestKnownBlock().getNumber()) {
+                    validatorMap.put(validator.getLeft(), validator.getRight());
+                    sendGetBlockHeaders(validator.getLeft(), 1, false);
```

Comment:
No, separate issue must be created for this as I referenced in description (reverted commit).